### DISCUSSION TITLE
onPress and Google Tag Manager

### DISCUSSION THREAD

**Discussion by pidupuis** [adobe/react-spectrum#4098] [Q&A]
The `onPress` method triggers an event for which the type is `press`. But it isn't catched by my Google Tag Manager who is catching clicks only.

How would you handle that?
Is there a way for GTM to catch `press` events? Or should I override the `onPress` event type to use `click` instead? Should I override the `onPress` method to dispatch a `click` event as well as the `press` event?

Thanks in advance for your help

**Comment by snowystinger** [comment]
Any chance you can use capture events for the click? we stop many of those events once we've handled them so we don't handle them again. https://www.simoahava.com/analytics/simple-custom-event-listeners-gtm/

Otherwise, you could wrap usages and fire a custom event for onPress/onPressStart/onPressEnd/onAction/etc, this might be useful because then you can actually track real interactions, not just click events, which sometimes the browser fires additional times for compatibility in certain scenarios.

Examples of where this might be useful
in a MenuTrigger, knowing when a menu has actually been opened. This is different on desktop and mobile (mouse down vs touch up, and the click event wouldn't actually be fired on the trigger in the desktop case as a result)
Menu items, knowing when pressing one has resulted in that action being taken.
